subsys/bluetooth/host/buf.c

```diff
@@ -64,7 +64,8 @@ static void iso_rx_freed_cb(void)
  * the HCI transport to fill buffers in parallel with `bt_recv`
  * consuming them.
  */
-NET_BUF_POOL_FIXED_DEFINE(sync_evt_pool, 1, SYNC_EVT_SIZE, 0, NULL);
+NET_BUF_POOL_FIXED_DEFINE(sync_evt_pool, CONFIG_BT_BUF_EVT_SYNCHRONOUS_COUNT, SYNC_EVT_SIZE, 0,
+			  NULL);
```

Comment:
```
/* Pool for RX HCI buffers that are always freed by `bt_recv`
 * before it returns.
 *
 * A singleton buffer shall be sufficient for correct operation.
 * The buffer count may be increased as an optimization to allow
 * the HCI transport to fill buffers in parallel with `bt_recv`
 * consuming them.
 */
```

Based on the comment in the implementation, and the fact that `bt_recv` control flow has implicit event flow control; I do not agree that this PR fixes any bug. Increasing the buffer count is an optimization at the cost of increased RAM usage, as each buffer can be over "quarter of a kilobyte" in size!

HCI Number of Completed Packets event "can" accumulate the counts for multiple handles and be generated at a frequency at the implementations discretion.